### PR TITLE
Bump version of `kube-state-metrics` to `v2.0.0-rc.1`

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

Bump version of some dependencies. `kube-state-metrics`, `datadog-operator`, `watermarkpodautoscaler`…

### Motivation

We need to bump the version of `kube-state-metrics` for the new `kubernetes_state_core` check.

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

The features the most impacted by this dependencies upgrade and that deserve further validation are:
* `kubernetes_state_core` core check
* `DatadogMetric`

### CODE REVIEW

File: go.mod
Comment (L3n41c) [inline go.mod:228]: TODO: Open a PR upstream and get rid of this reference to a personal fork.

File: pkg/collector/corechecks/cluster/orchestrator/orchestrator_transformer.go
Comment (L3n41c) [inline pkg/collector/corechecks/cluster/orchestrator/orchestrator_transformer.go:99]: In the new vendored version of `kubernetes`, the
```go
        // IPFamily is tombstoned to show why 15 is a reserved protobuf tag.
        // IPFamily *IPFamily `json:"ipFamily,omitempty" protobuf:"bytes,15,opt,name=ipFamily,Configcasttype=IPFamily"`
```
field has been replaced by:
```go
        IPFamilies []IPFamily `json:"ipFamilies,omitempty" protobuf:"bytes,19,opt,name=ipFamilies,casttype=IPFamily"`
```

I didn’t dare reflecting this structure change (`slice` instead of `pointer`) in `model.Service` as it might have an impact on the backend side.

Is concatenating the values like this OK ?

Comment (mfpierre) [reply pkg/collector/corechecks/cluster/orchestrator/orchestrator_transformer.go:99]: concatenating the values should be OK on our side 👍 

@L3n41c do you know how the backward compatibility is handled with previous kube versions?
Should we still check for the old field if the new field is empty?

File: go.mod
Comment (L3n41c) [reply go.mod:228]: Here is the upstream PR : kubernetes/kube-state-metrics#1455

**Review by olivielpeau — APPROVED** [review]
Approving for Agent Core, only reviewed the dep updates and I don't see any that affects the scope of Agent Core.

We'll want the CI to pass before merging this of course :)

Nit, not a blocker: could you add `/pkg/util/containerd/` to the list of directories owned by container-integrations? https://github.com/DataDog/datadog-agent/blob/9b2a63ab8df2c741cd24c8bc114ce29cafd129a0/.github/CODEOWNERS#L158-L162

File: pkg/clusteragent/externalmetrics/autoscaler_watcher_test.go
Comment (clamoriniere) [inline pkg/clusteragent/externalmetrics/autoscaler_watcher_test.go:32]: since the `scheme` is define in "package" not a test file. maybe it is better to move the init also in the "package"

File: pkg/clusteragent/externalmetrics/utils.go
Comment (clamoriniere) [inline pkg/clusteragent/externalmetrics/utils.go:104]:
```suggestion
func UnstructuredIntoDDM(obj interface{}, structDest *v1alpha1.DatadogMetric) error {
```

Comment (clamoriniere) [reply pkg/clusteragent/externalmetrics/utils.go:104]: maybe  you can also take directly an `*unstructured.Unstructured` as parameter

File: pkg/clusteragent/externalmetrics/utils.go
Comment (clamoriniere) [inline pkg/clusteragent/externalmetrics/utils.go:112]:
```suggestion
func UnstructuredFromDDM(structIn *v1alpha1.DatadogMetric, unstructOut *unstructured.Unstructured) error {
```

File: pkg/collector/corechecks/cluster/ksm/kubernetes_state.go
Comment (clamoriniere) [inline pkg/collector/corechecks/cluster/ksm/kubernetes_state.go:31]: 🥇

File: pkg/compliance/checks/kubeapiserver_check_test.go
Comment (clamoriniere) [inline pkg/compliance/checks/kubeapiserver_check_test.go:107]: you could move this function on top of the file

File: pkg/util/kubernetes/apiserver/util.go
Comment (clamoriniere) [inline pkg/util/kubernetes/apiserver/util.go:56]:
```suggestion
func UnstructuredFromWPA(structIn *v1alpha1.WatermarkPodAutoscaler, unstructOut *unstructured.Unstructured) error {
```

**Review by clamoriniere — COMMENTED** [review]
👏  for this contribution.
few same nits

File: pkg/collector/corechecks/cluster/orchestrator/orchestrator_transformer.go
Comment (L3n41c) [reply pkg/collector/corechecks/cluster/orchestrator/orchestrator_transformer.go:99]: The old field is just not available anymore: https://github.com/kubernetes/api/blob/v0.20.5/core/v1/types.go#L4210

File: pkg/clusteragent/externalmetrics/autoscaler_watcher_test.go
Comment (L3n41c) [reply pkg/clusteragent/externalmetrics/autoscaler_watcher_test.go:32]: In fact, this `scheme` is defined in a test file of the same package, in [`pkg/clusteragent/externalmetrics/datadogmetric_controller_test.go` here](https://github.com/DataDog/datadog-agent/pull/7857/files#diff-f5f5499a84c7aeb5c77b7fcad8a533bc8832a16c29f3bdfb7293eea7445e941cR32).

Inside `pkg/clusteragent/externalmetrics`, there are 2 test files:
* `autoscaler_watcher_test.go` and
* `datadogmetric_controller_test.go`

As they belong to the same package, there is only one `scheme` variable (which happens to be declared in `datadogmetric_controller_test.go`.
* `autoscaler_watcher_test.go` does the `AddToScheme(…)` for the HPA and the WPA and
* `datadogmetric_controller_test.go` does the `AddToScheme(…)` for the DDM.

In the end, there’s only one `scheme` variable for the tests only but I’ve scattered the `AddToScheme(…)` to the test file which is needing a given type. Does that make sense ?

File: pkg/clusteragent/externalmetrics/utils.go
Comment (L3n41c) [reply pkg/clusteragent/externalmetrics/utils.go]: :+1: Done: https://github.com/DataDog/datadog-agent/pull/7857/commits/067fca31ab65d678c7b2056c1a3438ed22b16113

File: pkg/clusteragent/externalmetrics/utils.go
Comment (L3n41c) [reply pkg/clusteragent/externalmetrics/utils.go]: :+1: Done: https://github.com/DataDog/datadog-agent/pull/7857/commits/067fca31ab65d678c7b2056c1a3438ed22b16113

File: pkg/compliance/checks/kubeapiserver_check_test.go
Comment (L3n41c) [reply pkg/compliance/checks/kubeapiserver_check_test.go]: :+1: Done: https://github.com/DataDog/datadog-agent/pull/7857/commits/067fca31ab65d678c7b2056c1a3438ed22b16113

File: pkg/util/kubernetes/apiserver/util.go
Comment (L3n41c) [reply pkg/util/kubernetes/apiserver/util.go]: :+1: Done: https://github.com/DataDog/datadog-agent/pull/7857/commits/067fca31ab65d678c7b2056c1a3438ed22b16113

**Review by clamoriniere — APPROVED** [review]

**Review by mfpierre — APPROVED** [review]
LGTM for our files, we'll add a QA card to pay attention to service IP changes for this release

**Pull request merged**
